verticapy/connection/connect.py

```diff
@@ -125,6 +129,52 @@ def connect(section: str, dsn: Optional[str] = None) -> None:
         prev_conn.close()
     try:
         gb_conn.set_connection(vertica_connection(section, dsn), section, dsn)
+        print("Connected Successfuly!")
+    except OAuthTokenRefreshError as e:
+        error_message = str(e)
+
+        from verticapy.connection.write import new_connection
+
+        # Check if the error message contains the refresh token part
+        if "Failed getting OAuth access token from a refresh token" in error_message:
```

Comment:
I have a slight concern with checking for a particular error message. It could conceivably change in vertica-python and VerticaPy would be unaware. However vertica-python uses OAuthTokenRefreshError for missing client secret, refresh token, client id, etc as well. So checking the error message may be the best thing to do, and have a negative test associated with this as well.